echo-rest/src/main/groovy/com/netflix/spinnaker/echo/events/RestEventListener.groovy

```diff
@@ -51,15 +54,20 @@ class RestEventListener implements EchoEventListener {
       try {
         Map eventAsMap = mapper.convertValue(event, Map)
         Map sentEvent = eventAsMap
+
         if (service.config.flatten) {
           eventAsMap.content = mapper.writeValueAsString(eventAsMap.content)
           eventAsMap.details = mapper.writeValueAsString(eventAsMap.details)
         }
         if (service.config.wrap) {
-          sentEvent = [
-            "eventName": "${service.config.eventName ?: eventName}" as String,
-          ]
-          sentEvent["${service.config.fieldName ?: fieldName}" as String] = eventAsMap
+          if(service.config.template) {
+            sentEvent = restEventTemplate.render(service.config.template as String, sentEvent)
```

Comment:
Would it be more idiomatic to have the value of service.config.template built into restEventTemplate? maybe at the time of construction? Since the interface is called RestEventTemplate and the class is called SimpleEventTemplate, I would expect the template itself to be stored in the class rather than passed to the method.